mono/metadata/cominterop.c

```diff
@@ -643,11 +648,25 @@ void
 mono_mb_emit_cominterop_get_function_pointer (MonoMethodBuilder *mb, MonoMethod *method)
 {
 #ifndef DISABLE_JIT
+	int slot;
+	MonoError error;
 	// get function pointer from 1st arg, the COM interface pointer
 	mono_mb_emit_ldarg (mb, 0);
-	mono_mb_emit_icon (mb, cominterop_get_com_slot_for_method (method));
-	mono_mb_emit_icall (mb, cominterop_get_function_pointer);
-	/* Leaves the function pointer on top of the stack */
+	slot = cominterop_get_com_slot_for_method (method, &error);
+	if (is_ok (&error)) {
+		mono_mb_emit_icon (mb, slot);
+		mono_mb_emit_icall (mb, cominterop_get_function_pointer);
+		/* Leaves the function pointer on top of the stack */
+	}
+	else {
+		/*
+		 * This code should never be executed and all invalid COM objects should be sieved out during their creation.
+		 * However, it will be emitted as a safeguard for cases when a method tries to interact with an invalid COM object from an unreachable basic block.
+		 */
+		char *msg = g_strdup_printf ("Unable to locate the interface on which method %s is defined.", mono_method_get_full_name (method));
```

Comment:
You are completely missing the point of MonoError: you are setting the error in `cominterop_get_com_slot_for_method` only not to use it here. Please use the exception from the MonoError instead of creating a new one from scratch.

Reply:
I'm not sure that is the case here.
The error created inside `cominterop_get_com_slot_for_method` signifies that the COM type is invalid, which means there's something wrong with the program being executed by Mono. That is "normal" in the sense that even though the program has an invalid COM type, the runtime has spotted it timely and predictably. Thus, an error generated within `cominterop_get_com_slot_for_method` indicates there's something wrong with the program being executed, but not the runtime. 
This line, however, generates a safeguard that effectively marks this basic block as "bad" because we've deduced that a method on an invalid COM object is invoked from it. An exception should've been thrown during the creation of an invalid COM object, so this block should not have been reached. If we do find ourselves here, that means something went wrong in the runtime, which sounds like a case of `ExecutionEngineException` to me. I use this exception as an assertion here.
Perhaps something is wrong with my reasoning, in which case I'll be grateful if you point out what exactly is wrong. Admittedly I've not used that mechanism much. 
The way I see it right now, it would be logically wrong to use the same kind of `MonoError` in those two functions, and that is the reason why my initial patch had a simple boolean error code instead of `MonoError` in the first place.

Reply:
You can just use the `InvalidProgramException`.

Reply:
I found out that CLR throws `InvalidOperationException`, so I'll update the patch to emit that one instead.
Furthermore, I found a case where this stub will be reached under "normal" circumstances (namely, if the incomplete method of a ComImport class is static) rendering my original presumption ("this should never be reached") incorrect. Thus, the exception does need to be the same and my previous.
I'll update the PR soon with these changes.